pkg/pxc/app/statefulset/haproxy_test.go

```diff
@@ -108,3 +108,138 @@ func pointerToTrue() *bool {
 	b := true
 	return &b
 }
+
+func TestHAProxyHealthCheckEnvVars(t *testing.T) {
+	tests := map[string]struct {
+		healthCheck     *api.HAProxyHealthCheckSpec
+		expectedEnvVars map[string]string
+	}{
+		"default values": {
+			healthCheck: nil,
+			expectedEnvVars: map[string]string{
+				"HA_SERVER_OPTIONS": "resolvers kubernetes check inter 10000 rise 1 fall 2 weight 1",
+			},
+		},
+		"custom interval only": {
+			healthCheck: &api.HAProxyHealthCheckSpec{
+				Interval: func() *int32 { i := int32(3000); return &i }(),
+			},
+			expectedEnvVars: map[string]string{
+				"HA_SERVER_OPTIONS": "resolvers kubernetes check inter 3000 rise 1 fall 2 weight 1",
+			},
+		},
+		"custom fall only": {
+			healthCheck: &api.HAProxyHealthCheckSpec{
+				Fall: func() *int32 { i := int32(3); return &i }(),
+			},
+			expectedEnvVars: map[string]string{
+				"HA_SERVER_OPTIONS": "resolvers kubernetes check inter 10000 rise 1 fall 3 weight 1",
+			},
+		},
+		"custom rise only": {
+			healthCheck: &api.HAProxyHealthCheckSpec{
+				Rise: func() *int32 { i := int32(2); return &i }(),
+			},
+			expectedEnvVars: map[string]string{
+				"HA_SERVER_OPTIONS": "resolvers kubernetes check inter 10000 rise 2 fall 2 weight 1",
+			},
+		},
+		"all custom values": {
+			healthCheck: &api.HAProxyHealthCheckSpec{
+				Interval: func() *int32 { i := int32(3000); return &i }(),
+				Fall:     func() *int32 { i := int32(2); return &i }(),
+				Rise:     func() *int32 { i := int32(1); return &i }(),
+			},
+			expectedEnvVars: map[string]string{
+				"HA_SERVER_OPTIONS": "resolvers kubernetes check inter 3000 rise 1 fall 2 weight 1",
+			},
+		},
+		"shutdown on mark down enabled": {
+			healthCheck: &api.HAProxyHealthCheckSpec{
+				ShutdownOnMarkDown: true,
+			},
+			expectedEnvVars: map[string]string{
+				"HA_SERVER_OPTIONS":        "resolvers kubernetes check inter 10000 rise 1 fall 2 weight 1",
+				"HA_SHUTDOWN_ON_MARK_DOWN": "yes",
+			},
+		},
+		"all custom values with shutdown": {
+			healthCheck: &api.HAProxyHealthCheckSpec{
+				Interval:           func() *int32 { i := int32(3000); return &i }(),
+				Fall:               func() *int32 { i := int32(2); return &i }(),
+				Rise:               func() *int32 { i := int32(1); return &i }(),
+				ShutdownOnMarkDown: true,
+			},
+			expectedEnvVars: map[string]string{
+				"HA_SERVER_OPTIONS":        "resolvers kubernetes check inter 3000 rise 1 fall 2 weight 1",
+				"HA_SHUTDOWN_ON_MARK_DOWN": "yes",
+			},
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			cr := &api.PerconaXtraDBCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-cluster",
+				},
+				Spec: api.PerconaXtraDBClusterSpec{
+					CRVersion: "1.18.0",
```

Comment:
please use `version.Version()` instead of hardcoding the version, this will fix unit test failures.